compiler/src/main/kotlin/se/ansman/kotshi/FactoryProcessingStep.kt

```diff
@@ -75,10 +75,10 @@ class FactoryProcessingStep(
                                 genericAdapters.isEmpty() -> addCode(handleRegularAdapters(regularAdapters))
                                 regularAdapters.isEmpty() -> addCode(handleGenericAdapters(genericAdapters))
                                 else -> {
-                                    addIfElse("type instanceof \$T", ParameterizedType::class.java) {
+                                    addIf("type instanceof \$T", ParameterizedType::class.java) {
                                         addCode(handleGenericAdapters(genericAdapters))
                                     }
-                                    addElse { addCode(handleRegularAdapters(regularAdapters)) }
+                                    addCode(handleRegularAdapters(regularAdapters))
```

Comment:
No need for an `else` and nesting since every emitted conditional ends in `return`.